### PR TITLE
Paged search control

### PR DESCRIPTION
**Pull request by cwaldbieser**
I'm trying this PR again, but with the git commit history cleaned up a bit.
This adds support for the paged search control, which is important if you want to do LDAP searches that return lots of entries on AD.

### Contributor Checklist:

* [ x] I have updated the release notes at `docs/source/NEWS.rst` 
* [ x] I have updated the automated tests.
* [ x] All tests pass on your local dev environment. See `CONTRIBUTING.rst`.

### CODE REVIEW

**Comment by psi29a** [comment]
Almost there! :)

**Comment by psi29a** [comment]
I honestly think this is super anal-retentive. ;)

**Comment by cwaldbieser** [comment]
I'm not sure what's going on, but I'm having some issues reading the codedev report?  Now when I see it in github, it says merging will actually increase coverage by 0.24%.  And a few minutes ago I'm pretty sure it would decrease coverage-- but I didn't change anything.

**Comment by cwaldbieser** [comment]
I think I see where it dinged me.  I added some missing blank lines and it's decided that the surrounding code is part of my change and needs coverage.  I'll add some tests for those.

**Comment by cwaldbieser** [comment]
99.47% - at least I am on the right track.

**Comment by cwaldbieser** [comment]
Is `codecov` stuck?  Or does it sometimes just take a long time to report out?

**Comment by cwaldbieser** [comment]
I was able to force `coddev` to push notifications.  99.58%.  But now I can't see where the missing coverage in my diffs is.  @psi29a , is there some way we can force the checks to re-run?  I can add another docstring to a function, but that seems like a hokey-pokey way to fire off the CI toolchain.

**Comment by psi29a** [comment]
Can't you send a empty commit? that should trigger another run.

**Comment by cwaldbieser** [comment]
I didn't even realize you could make an empty commit.  Weird.  Hopefully this will be the final iteration.

**Comment by cwaldbieser** [comment]
It took me a while to finally figure out what the yellow "partials" were about, but I finally passed the tests!

File: ldaptor/test/test_ldapclient.py [hunk 1]
```diff
@@ -210,7 +210,7 @@ def collect_result_(result):
 
         client.send_multiResponse(op, collect_result_)
         expected_value = pureldap.LDAPMessage(op)
-        expected_value.id = 1
+        expected_value.id -= 1
```
Comment (psi29a) [inline ldaptor/test/test_ldapclient.py]: nice find :)

**Comment by psi29a** [comment]
Merged! Thank you!

**Pull request merged**